models/points.go

```diff
@@ -167,7 +167,10 @@ func ParsePointsWithPrecision(buf []byte, defaultTime time.Time, precision strin
 		pos++
 
 		if len(block) == 0 {
-			break
+			if pos >= len(buf) {
```

Comment:
Could you not move this `pos >= len(buf)` check up to line `168`, after `pos++`? Then you could remove the `pos >= len(buf)` check on line `200`.